ruby_scripts/common/cart_amount_qualifier.rb

```diff
@@ -19,7 +19,7 @@ def match?(cart, selector = nil)
         compare_amounts(cart.subtotal_price_was - @amount, @comparison_type, total)
       when :diff_item
         original_line_total = cart.line_items.reduce(Money.zero) do |total, item|
-          total + (selector&.match?(item) ? item.original_line_price : Money.zero)
+          total + (selector.nil? || selector.match?(item) ? item.original_line_price : Money.zero)
```

Comment:
I think this way makes more sense as it will consider the entire amount of the cart instead of adding up to 0 if no selector is given. Perhaps requiring a selector and `raise`ing makes more sense that unexpected behavior though 🤔